### ISSUE TITLE
Bug: Incorrect error message

### ISSUE THREAD

**Issue by toschlog** [facebook/react#21078]
I'm getting this error message:

  React Hook "useStyles" is called conditionally. React Hooks must be called in the exact same order in every component render

However, I'm pretty darn sure that I'm not actually calling useStyles conditionally.  Also, the error messages come and go if I comment out unrelated code.

React version: 17.0.1

## Steps To Reproduce

1. Attempt to run my app

Unfortunately this error only happens with one particular big source file.  I don't see an easy way to show it to you without sending my whole project.  I don't mind doing that, however.

**Comment by gaearon** [comment]
This is a lint error message, so you don't need to send the whole file. You just need to send the component in which this happens.

**Comment by toschlog** [comment]
Ok.  Can I submit a source file privately?  

I'm able to make the error go away by adding a lot of eslint-disable-next-line comments, so that's good.

**Comment by gaearon** [comment]
You can remove all code around until you have a minimal example.
https://stackoverflow.com/help/minimal-reproducible-example

**Comment by toschlog** [comment]
> You can remove all code around until you have a minimal example.

Not really.  I tried to do this, but I found that making completely unrelated changes made the error come and go.

For example, here are the last few lines of my file:

![image](https://user-images.githubusercontent.com/25090868/112375219-69fda100-8cb9-11eb-84e1-51c771225b5f.png)

This file gets a pile of hook errors.  However, if I uncomment-out the code at line 3654, the errors go away.  So I'm guessing the the errors have something to do with the size and complexity of the source file.

**Comment by gaearon** [comment]
You can send me the file at my username @fb.com if you’d prefer. But if it’s so strange then it might just be an ESLint bug.

**Comment by toschlog** [comment]
File sent.

It does sound like it could be a lint error.

**Comment by noahkrause** [comment]
I am also having this issue with a big source file. Messy, but gets the point across. Removing this code fires linting errors stating that I am conditionally calling all of the hooks used in my component.

![image](https://user-images.githubusercontent.com/25876649/113269501-82b41b00-92a6-11eb-87bb-564cfbca7230.png)
